src/pfe/file-watcher/server/src/projects/projectUtil.ts

```diff
@@ -428,8 +429,14 @@ async function executeBuildScript(operation: Operation, script: string, args: Ar
 
                     if (operation.projectInfo.projectType == "odo") {
                         const projectHandler = await projectExtensions.getProjectHandler(operation.projectInfo);
-                        const appBaseURL: string = await projectHandler.getAppBaseURL(projectID);
-                        projectInfo.appBaseURL = appBaseURL.trim();
+                        const projectinfo: ProjectInfo = operation.projectInfo;
+                        const appBaseURL: string = (await projectHandler.getAppBaseURL(projectID)).trim();
+                        const appName: string = (await projectHandler.getAppName(projectID)).trim();
+                        projectInfo.appBaseURL = appBaseURL;
+                        projectInfo.compositeAppName = appName;
+                        projectinfo.appBaseURL = appBaseURL;
+                        projectinfo.compositeAppName = appName;
```

Comment:
Do we need both `projectInfo` and `projectinfo`? In the current change they are both referencing the same object. If we really don't want to modify the original object then we should do `odoProjectInfo = Object.assign({}, projectInfo);

Reply:
It's very tricky, and it's another naming problem in our existing code. They are referencing different projects.

`projectInfo` uses `ProjectEvent` interface: https://github.com/eclipse/codewind/pull/623/files#diff-7597df0b79304bd2be18940009414316L339, and it's used for Portal socket event, I think change it from `projectInfo` to `projectEvent` makes more sense, but won't break more things still use it.

`projectinfo` uses `ProjectInfo` interface, and it's used for Turbine layer project info.

Reply:
Ideally I think, but may do that in separate improvement PR.
`projectEvent` uses `ProjectEvent` interface
`projectInfo` uses `ProjectInfo` interface

Reply:
You're right on both accounts. It is using the `projectEvent` type and it is *very* confusing haha. I think at a minimum for this change let's at least rename `projectinfo` to `odoProjectInfo` since it's only used in the scope of this if block.

Reply:
Sounds good, and yeah, I was walking the confusing path before you, haha